flang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -245,6 +245,24 @@ static void parseCodeGenArgs(Fortran::frontend::CodeGenOptions &opts,
 
   opts.AliasAnalysis = opts.OptimizationLevel > 0;
 
+  if (const llvm::opt::Arg *a =
+          args.getLastArg(clang::driver::options::OPT_mframe_pointer_EQ)) {
+    llvm::StringRef s = a->getValue();
+
+    if (!(s == "none" || s == "non-leaf" || s == "all")) {
+      const auto debugWarningId = diags.getCustomDiagID(
+          clang::DiagnosticsEngine::Error, "Frame pointer: %0");
```

Comment:
I couldn't find it - which one is it?

Reply:
An error will cause the compiler to crash - I can't see any `RUN` lines in [rame-pointer-forwarding.f90](https://github.com/llvm/llvm-project/pull/74598/files#diff-fad5151400801da3763ab30d3470d69a748b236f0793afc41a9a94f6432a7c78) that would cause the compiler to crash.

Also, please don't "resolve" threads that haven't been concluded yet - this makes reviewing harder.

Reply:
@banach-space the test is in func-attr.f90 https://github.com/llvm/llvm-project/pull/74598/files#diff-1d4cea32c615e677fc20e34ffd2f5c3e0f84b00f5af6a692d787747e17bd9e3d

```
! RUN: not %flang_fc1 -triple aarch64-none-none -mframe-pointer=wrongval -emit-llvm -o - %s 2>&1| FileCheck %s --check-prefix=CHECK-WRONGVALUEFP
```

Reply:
Now I see :)